src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java

```diff
@@ -197,20 +197,24 @@ public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher
         if (!matlabExecutablePath.exists()) {
             throw new MatlabNotFoundError(Message.getValue("matlab.not.found.error"));
         }
+        FilePath matlabBinDir = matlabExecutablePath.getParent();
+        if (matlabBinDir == null) {
+            throw new MatlabNotFoundError(Message.getValue("matlab.not.found.error"));
+        }
         // Add matlab-batch executable in path
         FilePath batchExecutable = getNthParentFilePath(matlabExecutablePath, 3);
         if (batchExecutable != null && batchExecutable.exists()) {
-            context.env("PATH+matlab_batch", batchExecutable.getRemote());
+            context.env("PATH+MATLAB_BATCH", batchExecutable.getRemote());
         }
 
         // Add "matlabroot" without bin as env variable which will be available across
         // the build.
         context.env("matlabroot", nodeSpecificMatlab);
         // Add matlab bin to path to invoke MATLAB directly on command line.
-        context.env("PATH+matlabroot", matlabExecutablePath.getParent().getRemote());
+        context.env("PATH+MATLAB_ROOT", matlabBinDir.getRemote());

```

Comment:
I'm not sure, should it be `MATLAB_ROOT` at line 212 as well?

Reply:
I'm not sure if it actually matters, but I ended up reverting this change anyways (back to matlabroot).